### ISSUE TITLE
Touchscreen: one press touches two buttons on two screens

### ISSUE THREAD

**Issue by eric-zhu** [kivy/kivy#7756]
**Software Versions**
* Python: 3.8
* OS: Ubuntu 18.04.4 LTS
* Kivy: 2.0.0
* Kivy installation method: pip install

**Describe the bug**
My kivy app has two screens: `menu` and `setting`. Both has a button to switch the screen to the other, i.e. `menu` -> `setting` -> `menu`.

When pressing the button from the `menu` screen, it looks nothing happened but the logs shows it switched to `setting` screen and then switch back to `menu` screen. It seems one press on touchscreen triggered two touch events: one on the `menu` screen and the other on the `setting` screen.

This only happened on touchscreen. No such an issue when clicking with a mouse.

**Expected behavior**
One press triggers one touch event. The screen is switched successfully.

**To Reproduce**
Run the example code and click `Goto settings` button.

**Code and Logs and screenshots**
```python
from kivy.app import App
from kivy.lang import Builder
from kivy.uix.screenmanager import ScreenManager, Screen, NoTransition

Builder.load_string("""
<MenuScreen>:
    BoxLayout:
        Button:
            id: b_menu
            text: 'Goto settings'
            on_press: root.manager.current = 'settings'
        Label:
            text: 'Quit'

<SettingsScreen>:
    BoxLayout:
        Button:
            id: b_setting
            text: 'Back to menu'
            on_press: root.manager.current = 'menu'
        Label:
            text: 'My settings button'
""")

# Declare both screens
class MenuScreen(Screen):
    def on_enter(self):
        print("Menu screen")

class SettingsScreen(Screen):
    def on_enter(self):
        print("Setting screen")

class TestApp(App):

    def build(self):
        # Create the screen manager
        sm = ScreenManager(transition=NoTransition())
        sm.add_widget(MenuScreen(name='menu'))
        sm.add_widget(SettingsScreen(name='settings'))

        return sm

if __name__ == '__main__':
    TestApp().run()
```
Kivy logs
```
[INFO   ] [Logger      ] Record log in /home/deploy/.kivy/logs/kivy_22-01-19_0.txt
[INFO   ] [Kivy        ] v2.0.0
[INFO   ] [Kivy        ] Installed at "/home/deploy/env/lib/python3.8/site-packages/kivy/__init__.py"
[INFO   ] [Python      ] v3.8.0 (default, Dec  9 2021, 17:53:27) 
[GCC 8.4.0]
[INFO   ] [Python      ] Interpreter at "/home/deploy/env/bin/python3"
[INFO   ] [Factory     ] 186 symbols loaded
[INFO   ] [Image       ] Providers: img_tex, img_dds, img_sdl2, img_pil (img_ffpyplayer ignored)
[INFO   ] [Window      ] Provider: sdl2
[INFO   ] [GL          ] Using the "OpenGL" graphics system
[INFO   ] [GL          ] Backend used <sdl2>
[INFO   ] [GL          ] OpenGL version <b'3.0 Mesa 20.0.8'>
[INFO   ] [GL          ] OpenGL vendor <b'Intel Open Source Technology Center'>
[INFO   ] [GL          ] OpenGL renderer <b'Mesa DRI Intel(R) Iris(R) Plus Graphics 655 (CFL GT3)'>
[INFO   ] [GL          ] OpenGL parsed version: 3, 0
[INFO   ] [GL          ] Shading version <b'1.30'>
[INFO   ] [GL          ] Texture max size <16384>
[INFO   ] [GL          ] Texture max units <32>
[INFO   ] [Window      ] auto add sdl2 input provider
[INFO   ] [Window      ] virtual keyboard not allowed, single mode, not docked
[INFO   ] [Text        ] Provider: sdl2
[INFO   ] [ProbeSysfs  ] device match: /dev/input/event6
[INFO   ] [MTD         ] Read event from </dev/input/event6>
[INFO   ] [Base        ] Start application main loop
[INFO   ] [MTD         ] </dev/input/event6> range position X is 0 - 32767
[INFO   ] [MTD         ] </dev/input/event6> range position Y is 0 - 32767
[INFO   ] [MTD         ] </dev/input/event6> range touch major is 0 - 0
[INFO   ] [GL          ] NPOT texture support is available
[INFO   ] [MTD         ] </dev/input/event6> range touch minor is 0 - 0
[INFO   ] [MTD         ] </dev/input/event6> range pressure is 0 - 255
[INFO   ] [MTD         ] </dev/input/event6> axes invertion: X is 0, Y is 0
[INFO   ] [MTD         ] </dev/input/event6> rotation set to 0
```

**Additional context**
Kivy config
```
[input]
mouse = mouse
mtdev_%(name)s = probesysfs,provider=mtdev
```

**Comment by RobertFlatt** [comment]
I have seen similar (https://github.com/kivy/kivy/issues/7707), events are blocked during a screen transition, but not after. If event handling is delayed and the button on the new screen is located in the same place, the new button sees the delayed event and triggers a second transition.

In this case I'd try removing 'mouse = mouse', as that might be giving double events.

Alternatively filter as I did. In my case a touch pad two finger horizontal move generated a storm of new mouse events (probably should be one touch start event with multiple move events).  I blocked these events for a timeout after transition using global flags (yuck!) https://github.com/Android-for-Python/gestures4kivy/blob/main/src/gestures4kivy/commongestures.py#L320-L352 . Of course the timeout is user detectable :(

Workarounds aside, this seems to me like an issue.
